keps/sig-architecture/4402-go-workspaces/README.md

```diff
@@ -0,0 +1,374 @@
+
+N/A.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+N/A.
+
+###### Are there any tests for feature enablement/disablement?
+
+N/A.
+
+### Rollout, Upgrade and Rollback Planning
+
+N/A.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+N/A.
+
+###### What specific metrics should inform a rollback?
+
+If there is massive downstream breakage, we may need to revert. If we decide
+to delay Go 1.22 and go back to 1.21, we WILL need to revert.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+Yes.  The CLI of various code-gen tools has to change.  These tools are used by
+downstream developers.
+
+### Monitoring Requirements
+
+N/A.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+N/A.
+
+###### How can someone using this feature know that it is working for their instance?
+
+N/A.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A.
+
+### Dependencies
+
+Go 1.22
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+N/A.
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+N/A.
+
+###### What are other known failure modes?
+
+None.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A.
+
+## Implementation History
+
+* Started work in 2022, needed more from Go team.
+* Work went stale.
+* Restarted in 2023.
+* KEP.
+
+## Drawbacks
+
+None.
+
+## Alternatives
+
+Change how we manage our mono-repo, abandon staging, or do something more
+dramatic wrt staging.
```

Comment:
Worth calling out as alternative, but staging as it stands today is clearly useful, so I don't think an alternative that abandons or replaces it would be viable.